applications/nrf_desktop/doc/config_channel.rst

```diff
@@ -61,37 +98,256 @@ Setting a configuration value of a device, forwarded by the dongle to a paired d
+
+If you develop a custom host tool, use the Recipient ID linked with the device that is being discovered.
+Your tool should note which Recipient ID and HID instance on the host is associated with the board name and Hardware ID.
+
+.. _nrf_desktop_config_channel_module_discovery:
+
+Module discovery
+----------------
+
+The host can access options provided by the application modules.
+Before it gets access to the options, the host must identify which modules are available and what options are provided by them.
+
+Both modules and their options are identified by a name, sent as a string during discovery procedure:
+
+* The module of the given name is linked with a specific value of Module ID.
+* The option of the given name is linked with a specific value of Option ID.
+
+The Module ID and Option ID associated with the module and the option name can vary between devices, but they do not change between connections to the same device and firmware version.
+
+If you develop a custom host tool, you need to ensure your tool performs the following steps for the module discovery procedure:
+
+1. Obtain the number of configurable application modules using the ``CONFIG_STATUS_GET_MAX_MOD_ID`` request.
+   The response returns the highest value of Module ID available on the device.
+#. Read the module descriptor of every application module (iterate Module ID from ``0`` up to and including the maximum supported Module ID).
+   Implement the following order:
+
+   a. The module descriptor is read using fetch request with the current Module ID while Option ID is set to ``0``.
+      The transaction must be performed multiple times, with responses for subsequent requests containining strings that are part of the module descriptor.
+      The end line character (``\n``) should indicate the end of the descriptor.
+      After the end line character is fetched, subsequent requests should loop around and repeat descriptor strings.
+      The module descriptor strings are provided in a predefined order, but the host should make no assumption about the descriptor string that will be provided by the device as the first one.
+      The duplicated string is fetched after host received all of the strings related to given module.
+      The host can stop option discovery procedure when duplicated string is fetched.
+   #. The module name should be provided as the first string in the module descriptor.
+   #. The subsequent strings should indicate the module option names.
+      The first option on the descriptor must be identified with Option ID equal to ``1``.
+      Subsequent options will be identified by monotonically increasing Option ID values (second option by ``2``, third by ``3``, and so on).
+   #. Once the descriptor is read, the module options can be accessed.
+      When performing set or fetch request on the option, the Event ID contains Option ID and Module ID of module owning the option.
```

Comment:
This is very hard to read. We need to rephrase in a separate commit, but I'm going to do this after this review, when I have a second between the meetings.